packager/media/formats/mp4/mp4_muxer.cc

```diff
@@ -293,6 +293,11 @@ Status MP4Muxer::DelayInitializeMuxer() {
       entry.media_time = edit_list_offset_.value();
       entry.media_rate_integer = 1;
       trak.edit.list.edits.push_back(entry);
+    } else if (stream->media_time() != 0) {
+      EditListEntry entry;
```

Comment:
Not sure if its easily feasible, do you think having this logic isolated in the `MP4InitMuxer` could be done?

Reply:
I've considered that as well. Due to the invocation of `FireOnMediaStartEvent()` at the end of this function, we have a couple of alternatives: 1) We could make this function virtual and override it in `MP4InitMuxer`, though it would necessitate duplicating the existing code; 2) Alternatively, we could introduce a virtual member function in `MP4Muxer` specifically for managing the moov box, which can then be overridden in `MP4InitMuxer` to include this logic. The current approach minimizes changes, but I am open to additional suggestions.

Reply:
Do you know why original code from UpdateEditListOffsetFromSample is not setting edit_list_offset_ as it do for initial shaka packaging ?

Reply:
In `UpdateEditListOffsetFromSample()`, `edit_list_offset_` is computed as the PTS-DTS difference, which requires having access to the first sample.  In our use case, we don't have that information when we're packaging only the init segment.

Reply:
hmmm, ok. sounds correct.